servlet/src/main/java/com/linecorp/armeria/server/servlet/ServletRegistration.java

```diff
@@ -0,0 +1,252 @@
+     */
+    @Nullable
+    public ServletSecurityElement getServletSecurityElement() {
+        return servletSecurityElement;
+    }
+
+    /**
+     * Get multipart config element.
+     */
+    @Nullable
+    public MultipartConfigElement getMultipartConfigElement() {
+        return multipartConfigElement;
+    }
+
+    /**
+     * Get servlet config.
+     */
+    public ServletConfig getServletConfig() {
+        return servletConfig;
+    }
+
+    /**
+     * Get servlet.
+     */
+    public Servlet getServlet() {
+        return servlet;
+    }
+
+    /**
+     * Is async supported.
+     */
+    public Boolean isAsyncSupported() {
+        return asyncSupported;
+    }
+
+    /**
+     * Get load on startup.
+     */
+    public int getLoadOnStartup() {
+        return loadOnStartup;
+    }
+
+    /**
+     * Is initialization servlet cas.
+     */
+    public boolean isInitServletCas(boolean expect, boolean update) {
+        return initServlet.compareAndSet(expect, update);
+    }
+
+    /**
+     * Is initialization servlet.
+     */
+    public boolean isInitServlet() {
+        return initServlet.get();
+    }
+
+    @Override
+    public Set<String> addMapping(String... urlPatterns) {
+        requireNonNull(urlPatterns, "urlPatterns");
+        mappingSet.addAll(Arrays.asList(urlPatterns));
+        for (String pattern : urlPatterns) {
+            urlMapper.addMapping(pattern, this, servletName);
+        }
+        return mappingSet;
+    }
+
+    @Override
+    public Collection<String> getMappings() {
+        return mappingSet;
+    }
+
+    @Override
+    @Nullable
+    public String getRunAsRole() {
+        return roleName;
+    }
+
+    @Override
+    public String getName() {
+        return servletName;
+    }
+
+    @Override
+    public String getClassName() {
+        return servlet.getClass().getName();
+    }
+
+    @Override
+    public boolean setInitParameter(String name, String value) {
+        throw new IllegalStateException("Can't set init parameter after ServletRegistration is initialized");
+    }
+
+    @Override
+    @Nullable
+    public String getInitParameter(String name) {
+        requireNonNull(name, "name");
+        return initParameterMap.get(name);
+    }
+
+    @Override
+    public Set<String> setInitParameters(Map<String, String> initParameters) {
+        throw new IllegalStateException("Can't set init parameters after ServletRegistration is initialized");
+    }
+
+    @Override
+    public Map<String, String> getInitParameters() {
+        return initParameterMap;
+    }
+
+    @Override
+    public void setLoadOnStartup(int loadOnStartup) {
+        this.loadOnStartup = loadOnStartup;
+    }
+
+    @Override
+    public Set<String> setServletSecurity(ServletSecurityElement constraint) {
+        requireNonNull(constraint, "constraint");
+        servletSecurityElement = constraint;
+        return new HashSet<>(servletSecurityElement.getMethodNames());
```

Comment:
ditto:
```java
* @return the (possibly empty) Set of URL patterns that were already
* the exact target of a <code>security-constraint</code> that was
* established via the portable deployment descriptor. This method
* has no effect on the patterns included in the returned set
```

Reply:
I think returning `HashSet<>(servletSecurityElement.getMethodNames())` for checking.